src/main/java/org/folio/rest/impl/TenantRefAPI.java

```diff
@@ -81,50 +83,85 @@ String servicePointUserFilter(String s) {
+
+    /**
+     * The module version we upgrade from, null if installing from scratch
+     */
+    public ModuleFrom(String moduleFrom) {
+      if (moduleFrom == null) {
+        semVer = null;
+      } else if (Character.isDigit(moduleFrom.charAt(0))) {
+        semVer = new SemVer(moduleFrom);
+      } else {
+        semVer = new ModuleId(moduleFrom).getSemVer();
+      }
+    }
+
+    /**
+     * Do we need to install a feature that has been introduced with the given version?
+     */
+    public boolean isNew(String version) {
+      if (semVer == null) {
+        return true;
+      }
+      return semVer.compareTo(new SemVer(version)) < 0;
+    }
+  };
+
   @Validate
   @Override
   Future<Integer> loadData(TenantAttributes attributes, String tenantId,
                            Map<String, String> headers, Context vertxContext) {
+
+    var moduleFrom = new ModuleFrom(attributes.getModuleFrom());
+
     // create topics before loading data
-    return
-      new KafkaAdminClientService(vertxContext.owner())
+    Future<Integer> future = new KafkaAdminClientService(vertxContext.owner())
         .createKafkaTopics()
-        .compose(x ->super.loadData(attributes, tenantId, headers, vertxContext))
-        .compose(superRecordsLoaded -> {
-          try {
-            List<URL> urls = TenantLoading.getURLsFromClassPathDir(
-              REFERENCE_LEAD + "/service-points");
-            servicePoints = new LinkedList<>();
-            for (URL url : urls) {
-              InputStream stream = url.openStream();
-              String content = IOUtils.toString(stream, StandardCharsets.UTF_8);
-              stream.close();
-              servicePoints.add(new JsonObject(content));
-            }
-          } catch (URISyntaxException | IOException ex) {
-            return Future.failedFuture(ex.getMessage());
-          }
-          TenantLoading tl = new TenantLoading();
-
-          tl.withKey(REFERENCE_KEY).withLead(REFERENCE_LEAD);
-          tl.withIdContent();
-          for (String p : refPaths) {
-            tl.add(p);
-          }
-          tl.withKey(SAMPLE_KEY).withLead(SAMPLE_LEAD);
-          tl.add("instances", "instance-storage/instances");
-          tl.withIdContent();
-          tl.add("holdingsrecords", "holdings-storage/holdings");
-          tl.add("items", "item-storage/items");
-          tl.add("instance-relationships", "instance-storage/instance-relationships");
-          if (servicePoints != null) {
-            tl.withFilter(this::servicePointUserFilter)
-              .withPostOnly()
-              .withAcceptStatus(422)
-              .add("users", "service-points-users");
-          }
-          return tl.perform(attributes, headers, vertxContext, superRecordsLoaded);
-        }).compose(result -> runJavaMigrations(attributes, vertxContext, headers).map(result));
+        .compose(x ->super.loadData(attributes, tenantId, headers, vertxContext));
+
+    if (moduleFrom.isNew("20.0.0")) {
```

Comment:
Why is this check for the version necessary in order to decide to load reference and sample data in addition to invoking `super.loadData`?

Reply:
TenantLoading doesn't support PUT for tables where optimistic locking is enabled.
The isNew check ensures that TenantLoading is called on the initial tenant install but not during any upgrade. I assume that people upgrade from Iris or later so that the previous version is always >= 20.0.0 on upgrades and this data load task is skipped.
If upgrading from Honeysuckle should be supported we need to add an upgrade task for https://github.com/folio-org/mod-inventory-storage/commit/a2b152596a93b5924887e0b8bbf52317eba5a69d to TenantRefAPI.java.

Reply:
> TenantLoading doesn't support PUT for tables where optimistic locking is enabled.

Could this have been changed to avoid all modules that use RAML Module Builder and will support optimistic concurrency control to make similar changes?

> The isNew check ensures that TenantLoading is called on the initial tenant install but not during any upgrade. I assume that people upgrade from Iris or later so that the previous version is always >= 20.0.0 on upgrades and this data load task is skipped.

You are assuming that no sample or reference data loading is needed, irrespective of the parameters folks use, if it isn't a first enablement of a module version in this family?

> If upgrading from Honeysuckle should be supported we need to add an upgrade task for [a2b1525](https://github.com/folio-org/mod-inventory-storage/commit/a2b152596a93b5924887e0b8bbf52317eba5a69d) to TenantRefAPI.java.

Have you asked folks if this is needed?

Reply:
Using PUT overwrites the reference/sample data, but this is not what people want because they have altered the reference data.
In most cases existing reference data should not been altered. This is accomplished by skipping data loading when the version check shows that the data has already been loaded.
Therefore the TenantLoading PUT doesn't support overwriting with optimistic locking yet.

Reply:
>> TenantLoading doesn't support PUT for tables where optimistic locking is enabled.

> Could this have been changed to avoid all modules that use RAML Module Builder and will support optimistic concurrency control to make similar changes?

No, because we don't know whether fields that have been changed since the inital load should
* be kept or
* be overwritten with the inital values.

Therefore upgrading has always been the task of the module, not the task of RMB's methods for initial data loading.

Reply:
> This is accomplished by skipping data loading when the version check shows that the data has already been loaded.

How can a version check determine if the data has been loaded or not?